config/config.go

```diff
@@ -191,6 +196,87 @@ func (cfg Config) CallbackURL(path string, mergeParams ...url.Values) string {
 	return base.String()
 }
 
+//  MatchURL will compare two url strings and will return true if they match.
+func MatchURL(baseURL, testURL string) (bool, error) {
+	compareQueryValues := func(baseVal, testVal url.Values) bool {
+		for name := range baseVal {
+			if baseVal.Get(name) == testVal.Get(name) {
+				continue
+			}
+			return false
+		}
+		return true
+	}
+
+	addImplicitPort := func(u *url.URL) {
+		if strings.Contains(u.Host, ":") {
+			return
+		}
+		switch strings.ToLower(u.Scheme) {
+		case "http":
+			u.Host += ":80"
+		case "https":
+			u.Host += ":443"
+		}
+	}
+	base, err := url.Parse(baseURL)
+
+	if err != nil {
+		return false, err
+	}
+	test, err := url.Parse(testURL)
+
+	if err != nil {
+		return false, err
+	}
+
+	addImplicitPort(base)
+	addImplicitPort(test)
+
+	// use full host that includes port by default, i.e example.com:8080
+	baseHost := base.Host
+	testHost := test.Host
+
+	// host/port check
+	if !strings.EqualFold(baseHost, testHost) {
```

Comment:
no need for temp vars here
```suggestion

	// host/port check
	if !strings.EqualFold(base.Host, test.Host) {
```